conans/client/graph/graph_binaries.py

```diff
@@ -98,9 +98,9 @@ def _evaluate_cache_pkg(self, node, package_layout, pref, metadata, remote, remo
 
     def _evaluate_remote_pkg(self, node, pref, remote, remotes):
         remote_info = None
-        if remote:
+        if remotes.selected:
```

Comment:
This is breaking most of the binary-remote logic in Conan.

If no remote is explicitly selected, the location of the binary package is:
- The one defined from the binary package, if it was previously downloaded
- The one defined for the recipe, if the recipe is associated with a remote.

As you can see in the CI, a ton of tests are broken by this.

We need to understand better:
- If this is a regression, it seems it is not, the behavior has always been there
- I'd say that the binary is not really downloaded, but the server is actually queried for the information of it. That is what the ``conan info`` command does. The title might be a bit misleading.
- Disabling all remotes indeed should completely avoid contacting them, if true, that could be a bug, but not related to the ``remotes.selected`` specification of remote.